pkgs/native_assets_cli/lib/src/model/build_config.dart

```diff
@@ -47,9 +47,9 @@ class BuildConfig {
   Uri? get ar => _ar;
   late final Uri? _ar;
 
-  /// Preferred packaging method for library.
-  PackagingPreference get packaging => _packaging;
-  late final PackagingPreference _packaging;
+  /// Preferred linkMode method for library.
+  LinkModePreference get linkModePreference => _linkModePreference;
```

Comment:
We need a `LinkMode` for the build output, it needs to say whether it's static or dynamic.

I'm not sure that 3 variables in the config is nicer than one enum-like variable. It would require reading all, and it enables inconsistent state: `allowStaticLinking=false` + `preferredLinkMode=LinkMode.static`. So keeping an enum is better IMO.